analyses/tmb-vaf-longitudinal/util/function-create-corplot.R

```diff
@@ -0,0 +1,75 @@
+#'
+#' @param maf 
+#' @param timepoints_other_plot 
+#' @param timepoints_deceased_plot 
+#' @param sid 
+#'
+#' @return A data frame with each possible combination of timepoints and create VAF corplot
+#' @export
+#'
+#' @examples
+create_corplot <- function(maf, timepoints_other_plot, timepoints_deceased_plot, sid) {
+
+  # Split maf, create df, and rename VAF column based on time point
+  timepoint_df <- maf[which(maf$timepoints_other == timepoints_other_plot), ] %>%
+    select(cancer_group, gene_protein, Hugo_Symbol, VAF) 
+  colnames(timepoint_df)[colnames(timepoint_df) == "VAF"] <- timepoints_other_plot
+  
+  deceased_df <- maf[which(maf$timepoints_deceased == timepoints_deceased_plot), ] %>%
+    select(cancer_group, gene_protein, Hugo_Symbol, VAF)
+  colnames(deceased_df)[colnames(deceased_df) == "VAF"] <- timepoints_deceased_plot
+  
+  maf_join <- deceased_df %>%
+    full_join(timepoint_df, by = c("gene_protein", "Hugo_Symbol", "cancer_group"), relationship = "many-to-many") %>%
+    mutate(sym = ifelse(Hugo_Symbol %in% oncoprint_goi$oncoprint_goi, TRUE, FALSE)) 
+  
+  maf_join[is.na(maf_join)] <- 0
+
+  # Create timepoint label to use for group column 
+  # only for timepoints = Diagnosis, Progressive, Recurrence
+  timepoint <- str_extract(timepoints_other_plot, "[^_]+")
+
+  # Create group column
+  maf_join$group <-  ifelse(maf_join[, timepoints_other_plot] > 0 & maf_join[, timepoints_deceased_plot] > 0, "Common",
+                            ifelse(maf_join[, timepoints_other_plot] > 0, timepoint,
+                                   ifelse(maf_join[, timepoints_deceased_plot] > 0, "Deceased",
+                                          ifelse(maf_join[, timepoints_other_plot] == 0 & maf_join[, timepoints_deceased_plot] == 0, "Remove", "Other"))))
+  
+  maf_join <- maf_join %>%
+    filter(!group %in% c("Remove", "Other"))
+  
+  # Convert labels to display on the plot only the ones in the goi list
+  maf_join$gene_protein <- ifelse(maf_join$sym == TRUE, maf_join$gene_protein, "")
+  
+  # Reorder time points
+  maf_join$group <- factor(x = maf_join$group, levels = c(timepoint, "Deceased", "Common"))
+ 
+  # Add cancer_group information
+  cancer_group <- maf_join$cancer_group
+  
+  # make corplot reproducible
+  # add this to ensure that we plot the same data points when using the geom_text_repel function
+  set.seed(2023)
+  
+  # Plot corplot 
+  p <- print(ggplot(maf_join, aes_string(x = timepoints_other_plot, y = timepoints_deceased_plot, color = "group")) +
+               geom_point(size = 10, fill = 4, alpha = 1 / 6) +
+               scale_colour_manual(values = palette_df$hex_codes) + 
+               labs(title = paste(sid, timepoint, "vs Deceased VAF Corplot", sep = " "),
+                    subtitle = cancer_group) + 
+               geom_vline(xintercept = 0.1, linetype = "dashed") + # Add vertical intercept
+               geom_hline(yintercept = 0.1, linetype = "dashed") + # Add horizontal intercept line
```

Comment:
There are some cases where "Common" mutations fall below the dashed lines specified here, which give the impression that they are unique to one timepoints. Maybe you could set them to zero, or remove altogether?